frontend/src/metabase/query_builder/actions/core/core.js

```diff
@@ -225,7 +225,7 @@ export const apiUpdateQuestion = (question, { rerunQuery } = {}) => {
     const originalQuestion = getOriginalQuestion(getState());
     question = question || getQuestion(getState());
 
-    rerunQuery = rerunQuery || getIsResultDirty(getState());
+    rerunQuery = rerunQuery ?? getIsResultDirty(getState());
```

Comment:
To not re-run the model query when it's updated in this action. The problem is that `runQuestionQuery` depends on the `queryBuilderMode`, and we need to set it first before running the query. So the dataset editor calls this action with `rerunQuery: false`, sets the query builder mode, and then runs it manually.